### PR TITLE
Remove shared *.schema.json in spec tests

### PR DESCRIPTION
**Pull request by v-pegao**
Clean-up usage of TestData.schema.json in spec tests in order that each test is more self-contained and more readable.

[Find related workitem here](https://ceapex.visualstudio.com/Engineering/_workitems/edit/115502/)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/5482)

### CODE REVIEW

File: docs/specs/escape.yml [hunk 1]
```diff
@@ -135,6 +134,34 @@ inputs:
     Link to @a&b?displayProperty=description
     Link to @a&c
     Link to <xref:a%2A>
+  _themes/ContentTemplate/schemas/TestData.schema.json: |
+    {
+      "type": "object",
```
Comment (yufeih) [inline docs/specs/escape.yml]: remove unnecessary properties in this schema json to make the test case short and more readable. E.g., this case only needs:

```
{
    "properties": {
         "uid": { "contentType": "uid" }
    },
    "xrefProperties": ["name", "fullName", "description"]
}
```

Comment (v-pegao) [reply docs/specs/escape.yml]: seems not working, i'm checking the diff between expected and actual

Comment (v-pegao) [reply docs/specs/escape.yml]: @yufeih Updated, could you please help review? Let me know if I should do this for all `TestData.schema.json`

Comment (yufeih) [reply docs/specs/escape.yml]: Yep, do this for all other cases

Comment (v-pegao) [reply docs/specs/escape.yml]: Done. Could you please help rewiew?

File: docs/specs/xref.yml [hunk 2]
```diff
@@ -338,6 +350,20 @@ repos:
...
+          {
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+              "uid": { "contentType": "uid" },
+              "xref": { "contentType": "Xref" },
+              "name": { "type": "string" },
+              "fullName": { "type": "string" },
+              "description": { "type": "string" },
+              "$schema": { "type": "string" }
```
Comment (yufeih) [inline docs/specs/xref.yml]:
```suggestion
              ~~~"$schema": { "type": "string" }~~~
```

Comment (v-pegao) [reply docs/specs/xref.yml]: This change will cause test failure, why we need this change?

Comment (yufeih) [reply docs/specs/xref.yml]: Oops, I meant removing these unnecessary lines like ~"$schema": { "type": "string" }~ and ~"type": "object",~ and ~"additionalProperties": false,~

Comment (v-pegao) [reply docs/specs/xref.yml]: Updated. Sorry for missing this one.

**Pull request merged**